fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/TimeDriftConstraint.java

```diff
@@ -0,0 +1,215 @@
+        this.lowerGapSecs = lowerGapSecs;
+        this.upperGapSecs = upperGapSecs;
+    }
+
+    private static final Map<String, Long> SUPPORT_BOUND_FUNCTIONS = ImmutableMap.<String, Long>builder()
+            .put(FunctionSet.SECONDS_ADD, 1L)
+            .put(FunctionSet.MINUTES_ADD, 60L)
+            .put(FunctionSet.HOURS_ADD, 60L * 60L)
+            .put(FunctionSet.DAYS_ADD, 24L * 60L * 60L)
+            .put(FunctionSet.SECONDS_SUB, -1L)
+            .put(FunctionSet.MINUTES_SUB, -60L)
+            .put(FunctionSet.HOURS_SUB, -60L * 60L)
+            .put(FunctionSet.DAYS_SUB, -24 * 60L * 60L)
+            .build();
+    private static final String SUPPORT_BOUND_FUNCTION_NAMES = SUPPORT_BOUND_FUNCTIONS.keySet()
+            .stream().sorted().collect(Collectors.joining("/"));
+
+    public String getTargetColumn() {
+        return targetColumn;
+    }
+
+    public String getReferenceColumn() {
+        return referenceColumn;
+    }
+
+    public long getLowerGapSecs() {
+        return lowerGapSecs;
+    }
+
+    public long getUpperGapSecs() {
+        return upperGapSecs;
+    }
+
+    private static Pair<String, Long> parseBoundExpr(String spec, Expr expr) {
+        // days_sub is translated into TimestampArithmeticExpr in parse-time, so translated it back.
+        if ((expr instanceof TimestampArithmeticExpr) &&
+                ((TimestampArithmeticExpr) expr).getFuncName().equalsIgnoreCase(FunctionSet.DAYS_SUB)) {
+            expr = new FunctionCallExpr(FunctionSet.DAYS_SUB, expr.getChildren());
+        }
+        if (!(expr instanceof FunctionCallExpr)) {
+            throw new SemanticException(ERR_MSG_FORMAT, spec);
+        }
+        FunctionCallExpr fcall = (FunctionCallExpr) expr;
+
+        String fname = fcall.getFnName().getFunction();
+        if (!SUPPORT_BOUND_FUNCTIONS.containsKey(fname)) {
+            throw new SemanticException(ERR_MSG_FORMAT, spec);
+        }
+
+        if ((fcall.getChild(0) instanceof SlotRef) && (fcall.getChild(1) instanceof IntLiteral)) {
+            SlotRef column = (SlotRef) fcall.getChild(0);
+            IntLiteral delta = (IntLiteral) fcall.getChild(1);
+            String columnName = column.getColumnName();
+            long deltaValue = SUPPORT_BOUND_FUNCTIONS.get(fname) * delta.getLongValue();
+            return Pair.create(columnName, deltaValue);
+        } else {
+            throw new SemanticException(ERR_MSG_FORMAT, spec);
+        }
+    }
+
+    private static final String ERR_MSG_FORMAT = "Invalid time_drift_constraint: '%s', " +
+            "legal format is '<targetColumn> between <boundExpr1> and <boundExpr2>', boundExpr must be " +
+            SUPPORT_BOUND_FUNCTION_NAMES + "(<referenceColumn>, <num>), " +
+            "targetColumn and referenceColumn must be DATE/DATETIME type";
+
+    public static TimeDriftConstraint parseSpec(String spec) {
+        Expr expr = SqlParser.parseSqlToExpr(spec, SqlModeHelper.MODE_DEFAULT);
```

Comment:
what if parse failed(eg: illegal input spec)?

Reply:
when adding time_drift_constraint via create/alter table statement, analyzer will report error;
when planning query, time_drift_constraint is not used if it fails to parse.